assets/javascripts/kitten/components/cards/crowdfunding-card.js

```diff
@@ -0,0 +1,499 @@
+import Truncate from 'react-truncate'
+
+const Text = Radium(TextBase)
+const Title = Radium(TitleBase)
+const HorizontalStroke = Radium(HorizontalStrokeBase)
+const Marger = Radium(MargerBase)
+
+class CrowdfundingCardComponent extends Component {
+  removeCurrentFocus = () => {
+    document.activeElement.blur()
+  }
+
+  render() {
+    const {
+      imageProps,
+      avatarProps,
+      titleProps,
+      ownerTitle,
+      ownerDescription,
+      cardTitle,
+      cardSubTitle,
+      titlesMinHeight,
+      titleTruncate,
+      info1,
+      info2,
+      info3,
+      progress,
+      state,
+      loading,
+      ...others,
+    } = this.props
+
+    const Tag = this.props.href ? 'a' : 'div'
+
+    // TODO: Remove radiumConfig when it is fixed in Radium.
+    return (
+      <StyleRoot radiumConfig={{ userAgent: 'all' }}>
+        <Tag
+          { ...others }
+          style={ styles.card }
+          onClick={ this.removeCurrentFocus }
+        >
+          { this.renderImage() }
+          { this.renderHeader() }
+
+          <Marger
+            bottom="2"
+            style={ titlesMinHeight && styles.titles }
+          >
+            { this.renderTitle() }
+            { this.renderSubtitle() }
+          </Marger>
+
+          { this.renderInformations() }
+          { this.renderProgress() }
+          { this.renderState() }
+          { this.renderLoading() }
+        </Tag>
+      </StyleRoot>
+    )
+  }
+
+  renderImage() {
+    const {
+      alt,
+      ...imageProps,
+    } = this.props.imageProps
+
+    const imageClassName = classNames(
+      'k-Card__image',
+      imageProps.className,
+    )
+
+    return (
+      <Marger
+        className="k-Card__imageContainer"
+        style={ styles.imageContainer }
+      >
+        { !this.props.loading &&
+          <img
+            { ...imageProps }
+            alt={ alt || '' }
+            className={ imageClassName }
+            style={ { ...imageProps.style, ...styles.image } }
+          />
+        }
+      </Marger>
+    )
+  }
+
+  renderHeader() {
+    return (
+      <div style={ styles.header.grid }>
+        <Marger
+          top="1"
+          bottom="1"
+        >
+          <ButtonImage
+            tag="span"
+            img={ !this.props.loading && this.props.avatarProps }
+            withoutPointerEvents
+            style={ styles.header.avatar }
+          />
+        </Marger>
+
+        <div
+          style={ styles.header.owner }
+        >
+          <Text
+            tag="div"
+            size="micro"
+            weight="regular"
+            style={[ this.props.loading && styles.header.owner.titleLoading ]}
+          >
+            { !this.props.loading && parseHtml(this.props.ownerTitle) }
+          </Text>
+
+          <Text
+            tag="div"
+            size="micro"
+            weight="light"
+            style={
+              [ this.props.loading && styles.header.owner.descriptionLoading ]
```

Comment:
En général on ne mets pas d'espace après et avant les `[` `]`.

Reply:
D'ailleurs, est-ce qu'on ne peut pas passer directement `this.props.loading && styles.header.owner.descriptionLoading` sans array ?

Reply:
Ca ne fonctionne pas directement sans array. Je suis passé par des variables du coup.